include/benchmark/benchmark_api.h

```diff
@@ -493,8 +525,40 @@ class State {
         internal::ThreadManager* manager);
 
  private:
-  void StartKeepRunning();
-  void FinishKeepRunning();
+  friend class internal::StateIterator;
+
+  size_t GetBatch(size_t batch_size, bool atomic_batch) {
+    // atomic_batch == true means the batch size will always be batch_size, and
+    // iteration will stop at the first multiple of batch_size greater than or
+    // equal to max_iterations.
+    // atomic_batch == false means always iterate to max_iterations. The last
+    // batch will be smaller than batch_size if it does not go evenly into
+    // max_iterations.
+    assert(batch_size != 0);
```

Comment:
Yeah, moving to cc saw a small performance hit (~1ns). Though we could move the slow path of the last ten or so lines into a separate method if that makes things a little cleaner.